javascriptv3/example_code/bedrock-agent/actions/delete-agent.js

```diff
@@ -0,0 +1,44 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { fileURLToPath } from 'url';
+import { checkForPlaceholders } from "../lib/utils.js";
+
+import { BedrockAgentClient, DeleteAgentCommand } from '@aws-sdk/client-bedrock-agent';
+
+/**
+ * @typedef {Object} DeleteAgentCommandOutput
+ * @property {string|undefined} agentId
+ * @property {Object|undefined} agentStatus
+ */
+
+/**
+ * Deletes an Amazon Bedrock Agent.
+ *
+ * @param {string} agentId - The unique identifier of the agent to delete.
+ * @param {string} [region='us-east-1'] - The AWS region in use.
+ * @returns {DeleteAgentCommandOutput} An object containing the agent id and status.
+ */
+export const deleteAgent = async (agentId, region = 'us-east-1') => {
```

Comment:
This function doesn't need the `async` keyword. You can return `client.send` without the await.

Reply:
But then it would return the promise instead of the result, right? In all other (non-list) Bedrock Agents actions we return the result, not the promise.

Reply:
Looking at it again, I think you're right, it should at least be consistent across all wrappers. Either return a Promise or return the final lists or objects respectively. What do you think?

Reply:
Good catch. I'll make it consistent across all action wrappers. Either return the promise, or the final object.

Reply:
In either case you're actually returning a promise. Just two different syntactical ways of doing so.